### PR TITLE
[BACKEND][RPU] Add Rhino RPU backend (Huixi Intelligence)

### PR DESCRIPTION
**Pull request by xup16**
## Summary

  Adds the **RPU backend** to FlagTree (`triton_v3.6.x`). The Rhino RPU is the
  AI accelerator of **Huixi Intelligence** (https://www.rhino.auto/), a company
  building high-performance, low-power compute platforms and chips for embodied
  AI and autonomous driving (e.g. its publicly announced R1 SoC).

  The backend follows the layout of the existing vendor backends (`aipu`,
  `enflame`, `hcu`, `mthreads`, …): a self-contained `third_party/rpu/` tree plus
  a few surgical, opt-in hooks in the shared build. The shared **compile loop is
  untouched** — `python/triton/compiler/compiler.py` and
  `python/triton/backends/compiler.py` have zero diff; RPU integrates through the
  stock stage contract. The only shared-Python change is making `torch` import
  optional in `hint_manager` (RPU runs torch-free). Everything else is a
  build-side opt-in or an extension of an existing backend alternation list, so
  other backends are unaffected. (74 files changed, +25,517 / −5.)

  ## Backend overview

  - **Dialects** (`third_party/rpu/{include,lib}/`): `RPU` (executable IR),
    `RPUPlan` (subgraph plan IR), and `RPUTransforms` (recognizer + lowering
    passes).
  - **Python entry points** (`third_party/rpu/backend/`):
    `RPUBackend(BaseBackend)` with stages `ttir → rpuplan → rpuexec → rpurc →
    rpuinstr → rpubin`, plus `RPUDriver(DriverBase)` and `RPULauncher`.
  - **Toolchain (vendor-provided)**: a custom RPU LLVM toolchain (`bin/clang` +
    `bin/rpuasm`) and the `rhino-launch-kernel` runtime, configured via
    `RPU_LLVM_ROOT` / `RPU_LK_RUNNER`. See `third_party/rpu/README.md`.

  ## Testing

  - **Backend/contract suite** (toolchain-free): `pytest
    third_party/rpu/python/test/unit/test_rpu_backend.py` → **100 passed**
    (stage contract, recognizer / executable-lowering / fallback paths,
    metadata, pipeline events).
  - **Compile suite** (on hardware): `pytest third_party/rpu/python/test/unit`
    exercises the full `ttir → … → rpubin` pipeline, dialects, and per-op
    contracts; it needs the vendor RPU LLVM toolchain (`RPU_LLVM_ROOT`) and runs
    in CI on the board → **204 passed, 8 skipped**.
  - **On-board smoke**: `third_party/rpu/python/test/board/lk_board_smoke.py
    --require-board` compiles 11 ops (`add`, `mul`, `mul_add`, `add_mul`, `relu`,
    `maximum`, `gemm`, `softmax`, `reduce_sum_axis0/1`, `broadcast_add`),
    dispatches each on `/dev/rpu`, and compares against a numpy golden.
  - Verified end-to-end on RPU hardware via the CI workflow (fresh build + full
    unit suite + 11-op on-board smoke) — **Succeeded**.

### CODE REVIEW

File: python/triton/compiler/hint_manager.py [hunk 1]
```diff
@@ -87,7 +87,10 @@ def normalize_backend_name(name: str) -> str:
 def hint_get_flagtree_backend() -> str:
     detected_backend = ""
 
-    import torch
+    try:
+        import torch
+    except ImportError:
+        return ""
```
Comment (zhzhcookie) [inline python/triton/compiler/hint_manager.py]: Move `import torch` to L97 ?

File: .github/workflows/rpu3.6-build-and-test.yml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+# The RPU CI runner is a small x86 host that orchestrates the build/test on a
+# separate aarch64 RPU board over SSH. The board cannot be reached directly
+# from a generic runner, so every wiring detail (board address, login user,
+# SSH key) lives only in the runner's local ~/.ssh/config under the opaque
+# alias `rpuboard` -- nothing host-specific is committed here. The runner
+# checks the source out locally, mirrors it to the board, then drives a clean
+# build and the test suite on the board.
+jobs:
+  rpu-build-and-test:
+    runs-on: huixi
```
Comment (zhzhcookie) [inline .github/workflows/rpu3.6-build-and-test.yml]: Should be `runs-on: rpu3.6`

File: python/triton/compiler/hint_manager.py (hunk 1 above)
Comment (xup16) [reply python/triton/compiler/hint_manager.py]: 这样 106行的pass 需要改成 return ""，是否同意这样修改

Comment (zhzhcookie) [reply python/triton/compiler/hint_manager.py]: > 这样 106行的pass 需要改成 return ""，是否同意这样修改

OK

**Review by zhzhcookie — APPROVED** [review]
LGTM

**Review by Galaxy1458 — APPROVED** [review]
LGTM
